mobile/src/bundles/common/components/button/button.tsx

```diff
@@ -1,19 +1,95 @@
 import React from 'react';
-import { Pressable } from 'react-native';
+import { type PressableProps } from 'react-native';
+import Icon from 'react-native-vector-icons/MaterialIcons';
 
-import { Text } from '~/bundles/common/components/components';
+import { Pressable, Text } from '~/bundles/common/components/components';
 
+import { type ButtonType } from '../../enums/enums';
 import { styles } from './styles';
 
+type ButtonTypeName = keyof typeof ButtonType;
+type StyleRecord = Record<string, unknown>;
+
 type Properties = {
     label: string;
-    onPress: () => void;
-};
+    iconName?: string;
+    iconSize?: number;
+    buttonType?: ButtonTypeName;
+} & PressableProps;
+
+const iconDefaultSize = 32;
+
+const Button: React.FC<Properties> = ({
+    label,
+    style: pressableStyle,
+    iconName = 'add',
```

Comment:
```suggestion
    iconName ,
```
This field should not be required, you can add icon only if name is provided

Reply:
fix